### PR TITLE
Initialization of fields in new fracture cells in HydroFracture solver 

### PR DESCRIPTION
**Pull request by frankfeifan**
In this PR, we 

- [x] Move the code that initializes fields in new fracture cells from `TwoPointsFluxApproximation` to `HydroFractureSolver`

https://github.com/GEOS-DEV/integratedTests/pull/107

### CODE REVIEW

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp [hunk 1]
```diff
@@ -965,6 +973,245 @@ void HydrofractureSolver< POROMECHANICS_SOLVER >::setUpDflux_dApertureMatrix( Do
   } );
 }
 
+template< typename POROMECHANICS_SOLVER >
+void HydrofractureSolver< POROMECHANICS_SOLVER >::initializeNewFractureFields( DomainPartition & domain )
+{
+#if !defined(SET_CREATION_DISPLACEMENT)
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: can we take advantage of this PR to remove all these macros? We can have two different initialization strategies for now if we are still testing which one works best but let's guard it with  proper flag. 
Also, I don't think we saw that much benefit from initializing the displacement.

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp [hunk 2]
```diff
@@ -965,6 +973,245 @@ void HydrofractureSolver< POROMECHANICS_SOLVER >::setUpDflux_dApertureMatrix( Do
...
+  #endif
+
+  #if ALLOW_CREATION_MASS==0
+    arrayView1d< real64 > const dens = fractureSubRegion.getReference< array1d< real64 > >( "density_n" );
+  #endif
+
+  #if SET_CREATION_PRESSURE==1
+    arrayView1d< real64 > const fluidPressure_n = fractureSubRegion.getField< fields::flow::pressure_n >();
+    arrayView1d< real64 > const fluidPressure = fractureSubRegion.getField< fields::flow::pressure >();
+    // Set the new face elements to some unphysical numbers to make sure they get set by the following routines.
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: feels like we can do something better than this.

**Review by CusiniM — APPROVED** [review]
We can clean this up even further. For example, I don't think we need any of those macros. For now, can we just merge this so that it is at least clear where the initialization occurs?

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp [hunk 3]
```diff
@@ -965,6 +983,182 @@ void HydrofractureSolver< POROMECHANICS_SOLVER >::setUpDflux_dApertureMatrix( Do
...
+          }
+          if( zeroDisp )
+          {
+            aperture[newElemIndex] = 0;
+          }
+        }
+
+        if( this->getLogLevel() > 1 )
+        {
+          printf( "New elem index, init aper, init press = %4d, %4.2e, %4.2e \n",
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: GEOS_LOG?

**Review by paveltomin — APPROVED** [review]
looks good, definitely better than before and change in the right direction
thanks!

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp (hunk 1 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: Fully support to remove the macroses, they are leading to dead code which doesn't even compile

**Comment by paveltomin** [comment]
When are you planning to merge this PR?

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp (hunk 3 above)
Comment (CusiniM) [reply src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: yes, absolutely.

**Comment by CusiniM** [comment]
> When are you planning to merge this PR?

as soon as possible

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp (hunk 3 above)
Comment (CusiniM) [reply src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: honestly we should never output stuff inside a kernel but this kernel has to be on host anyways so I think it's fine.

**Pull request merged**